fabric-data-generation-api-v1/src/main/java/net/fabricmc/fabric/api/datagen/v1/FabricTextureMap.java

```diff
@@ -0,0 +1,58 @@
+package net.fabricmc.fabric.api.datagen.v1;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+import net.minecraft.util.Identifier;
+
+/**
+ * Constructor class for making Maps for BlockModel textures.
+ * {@see FabricBlockModelSupplier}
+ */
+
+public class FabricTextureMap {
```

Comment:
Could you explain the difference between this and vanilla TextureMap?

Reply:
The difference between the two appears to be that the `FabricTextureMap` will just return the raw HashMap because it is much easier to access that way, the vanilla `TextureMap` appears to have no direct way to actually get the entries from a separate class, the existence of `FabricTextureMap` is to make it easier without the use of double-bracket indentation for HashMaps, although initially it was just taking in a HashMap.

This is how we get the entries for a `FabricTextureMap`:
```java
HashMap<String, Identifier> textureMap = fabricTextureMap.get();
```

You can't get the entries from a `TextureMap` - in fact, you can't access any of the inputted data without at least having one bit of pre-existing data. I'm contemplating whether to just revert back to the HashMap being a parameter or not.

Reply:
couldn't this be a simple interface injection to add a method to do so?

Reply:
Yeah, it doesn't really matter to me which implementation we go for (HashMap parameter, interface injection, etc.) - I'm mainly focused on the `FabricBlockModelSupplier` and the `SimpleBlockStateSupplier`, the existence of `FabricTextureMap` isn't necessarily required, although helpful.